library/core/class.bbcode.php

```diff
@@ -128,7 +128,13 @@ function doQuote($bbcode, $action, $name, $default, $params, $content) {
         if (isset($params['url'])) {
             $url = trim($params['url']);
 
-            if (is_numeric($url)) {
+            if (preg_match('/(c|d)-(\d+)/', strtolower($url), $matches)) {
+                if ($matches[1] === 'd') {
+                    $url = "/discussion/$matches[2]";
+                } else {
+                    $url = "/discussion/comment/$matches[2]#Comment_$matches[2]";
```

Comment:
For the sake of readability, do you mind wrapping the variables here (and above) in braces?